### PR TITLE
[CINN] Add constrains for while op

### PR DESCRIPTION
**Pull request by JiaWenxuan**
### PR Category

CINN

### PR Types

Bug fixes

### Description

pcard-76996
This PR added equal constraints when the symbol of a shape remains the same during args and outputs, then the symbol in input equals to output and added great than one constraints of the args.

### CODE REVIEW

File: paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc [hunk 1]
```diff
@@ -786,6 +786,36 @@ bool WhileOp::InferSymbolicShape(
...
+  for (int i = 0; i < num_results(); ++i) {
+    const auto &input_i =
+        shape_analysis->GetShapeOrDataForValue(operand_source(i + 1)).shape();
+    const auto &output_i =
+        shape_analysis->GetShapeOrDataForValue(result(i)).shape();
+    const auto &args_i =
+        shape_analysis->GetShapeOrDataForValue(body_args[i]).shape();
+    if (input_i.size() != output_i.size() || output_i.size() != args_i.size()) {
+      continue;
+    }
```
Comment (zyfncg) [inline paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc]: 对应value的size是要求必须相同的

Comment (JiaWenxuan) [reply paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc]: done

File: paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc [hunk 2]
```diff
@@ -737,6 +737,17 @@ bool WhileOp::InferSymbolicShape(
     shape_analysis->SetShapeOrDataForValue(value, shape_data);
   }
 
+  // add GreaterThanOne constraint
+  for (size_t i = 1; i < num_operands(); ++i) {
+    const auto &input_i =
+        shape_analysis->GetShapeOrDataForValue(operand_source(i + 1)).shape();
```
Comment (zyfncg) [inline paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc:748]: 这里是不是要用args？

Comment (JiaWenxuan) [reply paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc:748]: done

File: paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc [hunk 3]
```diff
@@ -737,10 +737,21 @@ bool WhileOp::InferSymbolicShape(
...
 
+  // add GreaterThanOne constraint
+  const auto &body_args = block_args();
+  for (size_t i = 0; i < body_args.size(); ++i) {
+    const auto &args_i =
+        shape_analysis->GetShapeOrDataForValue(body_args[i]).shape();
+    for (const auto &dim_expr : args_i) {
+      if (shape_analysis->IsGreatThanOne(dim_expr)) {
+        shape_analysis->AddGreatThanOneCstr(dim_expr);
+      }
```
Comment (zyfncg) [inline paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc]: 这个逻辑看上去什么也没做

**Review by tc20042008 — DISMISSED** [review]

File: paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc (hunk 3 above)
Comment (JiaWenxuan) [reply paddle/fluid/pir/dialect/operator/ir/control_flow_op.cc]: done

**Review by zyfncg — DISMISSED** [review]

**Review by zyfncg — APPROVED** [review]

**Pull request merged**